File: Leetcode/Python3/191. Number of 1 Bits/solution.py
# https://leetcode.com/problems/number-of-1-bits/

class Solution:
    def hammingWeight(self, n: int) -> int:
        if n < 2:
            return n
        answer = 0
        while n > 1:    # Binary number calculation
            q, r = divmod(n, 2)
            n = q
            if r != 0:
                answer += 1
            if n == 1:
                answer += 1
        return answer


class Solution:
    def hammingWeight(self, n: int) -> int:
        return bin(n).count('1')


class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n:
            n &= n - 1
            count += 1
        return count

File: Leetcode/C++/191. Number of 1 Bits/solution.cpp
// https://leetcode.com/problems/number-of-1-bits/description/

#include <stdint.h>

using namespace std;

class Solution
{
public:
    int hammingWeight(uint32_t n)
    {
        int count = 0;
        while (n > 0)
        {
            count += n % 2;
            n >>= 1;
        }
        return count;
    }
};

class Solution // Use gcc compiler built-in function
{
public:
    int hammingWeight(uint32_t n)
    {
        return __builtin_popcount(n);
    }
};